python/lsst/sims/photUtils/BandpassDict.py

```diff
@@ -0,0 +1,331 @@
+    """
+    This class will wrap an OrderedDict of Bandpass instantiations.
+
+    Upon instantiation, this class's constructor will resample
+    the input Bandpasses to be on the same wavelength grid (defined
+    by the first input Bandpass).  The constructor will then calculate
+    the 2-D phiArray for quick calculation of magnitudes in all
+    Bandpasses simultaneously (see the member method calcMagListFromSed).
+    """
+
+    def __init__(self, bandpassList, bandpassNameList):
+        """
+        @param [in] bandpassList is a list of Bandpass instantiations
+
+        @param [in] bandpassNameList is a list of tags to be associated
+        with those Bandpasses
+        """
+        self._bandpassDict = OrderedDict()
+        for bandpassName, bandpass in zip(bandpassNameList, bandpassList):
+
+            if bandpassName in self._bandpassDict:
+                raise RuntimeError("The bandpass %s occurs twice in your input " % bandpassName \
+                                   + "to BandpassDict")
+
+            self._bandpassDict[bandpassName] = copy.deepcopy(bandpass)
+
+        dummySed = Sed()
+        self._phiArray, self._wavelenStep = dummySed.setupPhiArray(self._bandpassDict.values())
+        self._wavelen_match = self._bandpassDict.values()[0].wavelen
+        self._nBandpasses = len(self._bandpassDict)
+
+
+    def __getitem__(self, bandpass):
+        return self._bandpassDict[bandpass]
+
+
+    def __len__(self):
+        return len(self._bandpassDict)
+
+
+    def __iter__(self):
+        for val in self._bandpassDict:
+            yield val
+
+
+    def values(self):
+        return self._bandpassDict.values()
+
+
+    def keys(self):
+        return self._bandpassDict.keys()
+
+
+    def calcMagListFromSed(self, sedobj, indices=None):
+        """
+        Return a list of magnitudes for a single Sed object.
+
+        @param [in] sedobj is an Sed object
+
+        @param [in] indices is an optional list of indices indicating which bandpasses to actually
+        calculate magnitudes for.  Other magnitudes will be listed as 'None' (i.e. this method will
+        return as many magnitudes as were loaded with the loadBandpassesFromFiles methods; it will
+        just return nonsense for magnitudes you did not actually ask for)
+
+        @param [out] magList is a list of magnitudes in the bandpasses stored in self.bandpassDict
+        """
+
+        if sedobj.wavelen is not None:
+
+            # If the Sed's wavelength grid agrees with self._wavelen_match to one part in
+            # 10^6, just use the Sed as-is.  Otherwise, copy it and resample it onto
+            # self._wavelen_match
+            if len(sedobj.wavelen)!=len(self._wavelen_match) or \
+            not numpy.allclose(sedobj.wavelen, self._wavelen_match, atol=0.0, rtol=1.0e-6):
+                dummySed = Sed(wavelen=sedobj.wavelen, flambda=sedobj.flambda)
+                dummySed.resampleSED(wavelen_match=self._bandpassDict.values()[0].wavelen)
```

Comment:
I did not notice the force kwarg in sed.resampleSED.  I can adjust how force is use so that we just pass force=True in dummySed.resampleSED() and then avoid having to do the second check.